tools/feature_version_audit.py

```
@@ -0,0 +1,685 @@
+        seen = set()
+        for name, ver, commit in new_features:
+            key = (name, ver, commit)
+            if key in seen:
+                continue
+            seen.add(key)
+            meta = feature_meta_map.get(normalize_name(name))
+            missing = False
+            if not meta or (not meta['mod_link'] and not (meta and meta['is_core'])) or not meta['description'] or not meta['key_features']:
+                missing = True
+            def boldmeta(val):
+                return f"**{val}**" if missing and val != '' and val != '-' else val
```

Ruff (0.11.9)

Function definition does not bind loop variable `missing`

(B023)